massa-protocol-worker/src/tests/operations_scenarios.rs

```diff
@@ -649,24 +654,24 @@ async fn test_protocol_propagates_operations_only_to_nodes_that_dont_know_about_
 
             // Send the operation to protocol
             // it should propagate to the node because it isn't in the block.
-            let mut ops = Map::default();
-            ops.insert(operation_id_2, operation_2);
+            let mut ops = Set::default();
+            ops.insert(operation_id_2);
             protocol_command_sender
                 .propagate_operations(ops)
                 .await
                 .unwrap();
 
             match network_controller
                 .wait_command(1000.into(), |cmd| match cmd {
-                    cmd @ NetworkCommand::SendOperations { .. } => Some(cmd),
+                    cmd @ NetworkCommand::AskForOperations { .. } => Some(cmd),
                     _ => None,
                 })
                 .await
             {
-                Some(NetworkCommand::SendOperations { node, operations }) => {
-                    let id = operations[0].content.compute_id().unwrap();
-                    assert_eq!(id, operation_id_2);
-                    assert_eq!(nodes[0].id, node);
+                Some(NetworkCommand::AskForOperations { to_node, wishlist }) => {
+                    assert_eq!(wishlist.len(), 1);
+                    assert!(wishlist.contains(&operation_id_2));
+                    assert_eq!(nodes[0].id, to_node);
```

Comment:
Now the propagation will just ask the received batch and not "Send" it.

In follow up, we will test if the algorithm send correctly the operation after receiving the full operations.

Reply:
outdated